src/constraint/spec.ts

```diff
@@ -23,45 +23,42 @@ export class SpecConstraintModel extends AbstractConstraintModel {
     super(specConstraint);
   }
 
+    public hasAllRequiredProperties(specM: SpecQueryModel): boolean {
+      return every(this.constraint.properties, (prop) => {
+        if (prop === Property.MARK) {
+          return !isEnumSpec(specM.getMark());
+        }
+
```

Comment:
Also should still throw error if prop is not mark and not encoding property to warn us to implement the right behavior here.